iron-list.html

```diff
@@ -1390,11 +1390,14 @@
 
       forceUpdate = forceUpdate || this._scrollHeight === 0;
       forceUpdate = forceUpdate || this._scrollPosition >= this._estScrollHeight - this._physicalSize;
+      // Probably a bug that style.height is a string like '100px' and it is being compared to
+      // a number?
       forceUpdate = forceUpdate || this.grid && this.$.items.style.height < this._estScrollHeight;
 
       // Amortize height adjustment, so it won't trigger large repaints too often.
       if (forceUpdate || Math.abs(this._estScrollHeight - this._scrollHeight) >= this._optPhysicalSize) {
         this.$.items.style.height = this._estScrollHeight + 'px';
+        this.updateViewportBoundaries();
```

Comment:
@freshp86 In your example, the list's scrollTarget (#container) has `offsetHeight = 0`.  

The iron-list's `_render` runs with that state, it renders the initial count which is `3`. Then based on those 3 items, the content size is calculated and set by `_updateScrollerSize`. However, because `#container` has no size, setting the size on a descendant will also affect the size of `#container`. When you resize the browser, the list takes the new scroll target's `height` which is incorrect anyways. 

If we were updating the height as you proposed, the list will keep rendering nodes until the limit is reached. To see that, change your loop to: ` for (var i = 0; i < 600; i++)` and resize the browser window.

Whatever change we make will not be correct, and I do agree that it would be great to throw or indicate in the console that something is wrong. I just haven't found a good way to do the detection since in many cases, the list is in a container with `display=none`.